packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -404,63 +512,317 @@ const handler: AdaptingProxyHandler<ProxyTarget, EditableTree> = {
     },
 };
 
+/**
+ * A Proxy target, which together with a `fieldProxyHandler` implements a basic access to
+ * the nodes of {@link EditableField} by means of the cursors.
+ */
+class FieldProxyTarget extends BaseProxyTarget implements ArrayLike<UnwrappedEditableTree> {
+    public readonly fieldKey: FieldKey;
+    public readonly fieldSchema: FieldSchema;
+    private readonly primaryType?: TreeSchemaIdentifier;
+    private readonly primaryField?: FieldKey;
+
+    constructor(
+        context: ProxyContext,
+        fieldKey: FieldKey,
+        fieldSchema: FieldSchema,
+        cursor: ITreeSubscriptionCursor,
+        primaryType?: TreeSchemaIdentifier,
+    ) {
+        // allowed only for empty roots
+        if (cursor.state !== ITreeSubscriptionCursorState.Current) {
```

Comment:
Our APIs for dealing with roots are a mess right now. I'm going to see if I can clean them up a bit so you don't have to do workaround like this

Reply:
[12630](https://github.com/microsoft/FluidFramework/pull/12630) should help avoid the need to special case root fields.